### ISSUE TITLE
file:// URLs: Support for relative path

### ISSUE THREAD

**Issue by tomtom** [mpcjanssen/simpletask-android#394]
It is unclear to me how file:// URLs are interpreted. There should be a way to define a path relative to the todo file. (If there already is one, the help docs don't mention it.)

**Comment by mpcjanssen** [comment]
There is a root folder setting. file:// will only refer to local files. 

The next version will use file:// to refer to fully qualified local content and root:// as a file relative to the configured root folder.

Basically in the next version, sharing a file to simpletask will add a link that will open that file.

**Comment by mpcjanssen** [comment]
The root folder setting is /sdcard/ by default. I do suspect that the share everything to simpletask feature will make this much less used though.

**Comment by tomtom** [comment]
> The next version will use file:// to refer to fully qualified local 
> content and root:// as a file relative to the configured root folder.

Would you be willing to support something like 
file://$TODOS/notes/foo.md in order to remain compatibility with other 
apps. E.g. on the bash command line, an environment variable in the URL 
gets expanded.

**Comment by mpcjanssen** [comment]
I think in that case it will be easy to use root:// and replace that with $TODOS in bash.

**Comment by smichel17** [comment]
What's the use case for a root that isn't the folder where you todo.txt file is located?

It seems to me that if I want to use a different directory, I should use file:// and specify the full path.

If you agree, you could remove the "root folder" setting.

**Comment by mpcjanssen** [comment]
@smichel17 root:// is for situations where you want relative links. For example of you have all your productivity files in at the same relative organisation but on different phones.

**Comment by smichel17** [comment]
@mpcjanssen if you have all your productivity files in the same relative organization, that includes your todo.txt file.

That is, you could remove the root setting, automatically set root={folder containing todo.txt}. Yes, you might not always have your todo.txt file in the root, but you can still have paths with ../*

**Comment by mpcjanssen** [comment]
I agree this could be an easier option, however this doesn't work for the Dropbox version, because Dropbox doesn't have local files and showing a remote file is tricky.

**Comment by mpcjanssen** [comment]
in new version, file:// is absolute path root:// is relative to root. But it is less of an issue because you can just share the target file to Simpletask and the url will be correct.

**Comment by smichel17** [comment]
What if the dropbox version used a local file, too, that it also synced to dropbox? Practically, it wouldn't do much except for backup and maybe saving offline changes. But it would simplify the interface in this ticket, and be the first step towards unifying the two versions\*.

\*It seems weird to me, that there are two versions. Why not just one, with a setting where I can enable dropbox sync? If I want cloudless, I'll leave that setting off. If I want dropbox, I'll turn it on. FWIW, that might also be the first step towards supporting other providers than dropbox (eg, owncloud).

**Comment by mpcjanssen** [comment]
@smichel17 

Problem with unifying is that the cloudless version then will need NETWORK permission in its manifest. For a large percentage of cloudless users this then disqualifies this app for work use, because it could communicate to the internet.
Additionally, I can't include Dropbox for FDroid without sharing my API keys and making Cloudless partially closed source. 

These are the main reason why there are two completely separate versions.

**Comment by mpcjanssen** [comment]
there are two options, `file://` or any other local file URI which refers to a local file and `todo://` which refers to another todo file relative to the todo file. Having full file support for Dropbox is not going to happen any time soon.

**Comment by smichel17** [comment]
@mpcjanssen Is that for cloudless, dropbox, or both? Is it planned or implemented? Is it replacing root:// links or in addition to them?

**Comment by mpcjanssen** [comment]
@smichel17  that's for both. I forgot about the root:// option which also still exists.

There really are 3 options:

1. todo:// : open a todo file on the storage (Dropbox or cloudless) relative to the opened todo file
2. root:// : open a local file relative to a define root using an Intent
3. anything other URL (such as http:// or file://) : open the Url using an Intent
